dandiapi/api/models/asset.py

```diff
@@ -117,6 +132,34 @@ def save(self, *args, **kwargs):
         self._populate_metadata()
         super().save(*args, **kwargs)
 
+    def generate_metadata(self, version):
+        """
+        Generate the correct metadata for this asset given a version.
+
+        The route used to download an asset will be different depending on which version it belongs
+        to. Since the download url is included in the asset metadata, it must be injected rather
+        than stored. This method should always be used instead of asset.metadata.metadata.
+        """
+        # TODO use http://localhost:8000 for local deployments
+        download_url = 'https://api.dandiarchive.org' + reverse(
```

Comment:
We could configure this with an environment variable but that also seems messy. Not sure what to do here.